osu.Game/Skinning/Components/IconElement.cs

```diff
@@ -0,0 +1,160 @@
+        {
+            public BindableIcon(IconModel defaultValue)
+                : base(defaultValue)
+            {
+            }
+
+            public override void Parse(object? input, IFormatProvider provider)
+            {
+                // When IconElement is deserialised from the skin, the value is passed here in the form of JObject.
+                if (input is JObject jObject)
+                    base.Parse(jObject.ToObject<IconModel>(), provider);
+                else
+                    base.Parse(input, provider);
+            }
+        }
+
+        private partial class SettingsIconDropdown : SettingsDropdown<IconModel>
+        {
+            protected override OsuDropdown<IconModel> CreateDropdown() => new IconDropdown();
+
+            private partial class IconDropdown : DropdownControl
+            {
+                public IconDropdown()
+                {
+                    foreach (var icon in all_icons)
+                        AddDropdownItem(icon);
+
+                    AlwaysShowSearchBar = true;
+                }
+
+                protected override LocalisableString GenerateItemText(IconModel item)
+                {
+                    if (item.Icon.Family is OsuIcon.FONT_NAME or OsuIcon.LEGACY_FONT_NAME)
+                        return $"osu!/{item.Name}";
+
+                    return $"{item.Icon.Weight}/{item.Name}";
+                }
+
+                protected override DropdownMenu CreateMenu() => new IconDropdownMenu
+                {
+                    MaxHeight = 200,
+                };
+            }
+
+            private partial class IconDropdownMenu : DropdownControl.OsuDropdownMenu
+            {
+                protected override DrawableDropdownMenuItem CreateDrawableDropdownMenuItem(MenuItem item) => new IconDropdownMenuItem(item);
+            }
+
+            private partial class IconDropdownMenuItem : DropdownControl.OsuDropdownMenu.DrawableOsuDropdownMenuItem
+            {
+                protected new IconItemContent Content => (IconItemContent)base.Content;
+
+                public IconDropdownMenuItem(MenuItem item)
+                    : base(item)
+                {
+                    var dropdownItem = (DropdownMenuItem<IconModel>)item;
+                    Content.Icon = dropdownItem.Value.Icon;
+                }
+
+                protected override Drawable CreateContent() => new IconItemContent();
+
+                protected partial class IconItemContent : ItemContent
+                {
+                    private readonly SpriteIcon spriteIcon;
+
+                    public IconUsage Icon
+                    {
+                        get => spriteIcon.Icon;
+                        set => spriteIcon.Icon = value;
+                    }
+
+                    public IconItemContent()
+                    {
+                        Label.Padding = new MarginPadding { Left = 35 };
+
+                        AddInternal(spriteIcon = new SpriteIcon
```

Comment:
I'm 99.9% sure you'll want to `DelayedLoadWrapper` this else every icon is getting loaded *into GPU memory* **permanently**.